kubetest/anywhere.go

```diff
@@ -267,24 +271,79 @@ func (k *kubernetesAnywhere) DumpClusterLogs(localPath, gcsPath string) error {
 		return nil
 	}
 
-	// the e2e framework in k/k does not support the "kubernetes-anywhere" provider,
-	// while a valid provider is required by the k/k "./cluster/log-dump/log-dump.sh" script
-	// for dumping the logs of the GCE cluster that kubernetes-anywhere creates:
-	//   https://github.com/kubernetes/kubernetes/blob/master/cluster/log-dump/log-dump.sh
-	// this fix is quite messy, but an acceptable workaround until "anywhere.go" is removed completely.
-	//
-	// TODO(neolit123): this workaround can be removed if defaultDumpClusterLogs() is refactored to
-	// not use log-dump.sh.
-	providerKey := "KUBERNETES_PROVIDER"
-	oldValue := os.Getenv(providerKey)
-	if err := os.Setenv(providerKey, "gce"); err != nil {
+	privateKeyPath := os.Getenv("JENKINS_GCE_SSH_PRIVATE_KEY_FILE")
+	if privateKeyPath == "" {
+		return fmt.Errorf("JENKINS_GCE_SSH_PRIVATE_KEY_FILE is empty")
```

Comment:
so are we just only supporting this in CI now? (seems ok, unsure if anyone else used this)

Reply:
i guess so, that's why i hardcoded the path originally, but i think we just want to patch this at this point.

we don't know if there are users for this deployer, but they shouldn't use it and we can just remove it soon (hopefully).
k-a as a repo may have some users, but it's beyond deprecated at this point.

Reply:
SGTM